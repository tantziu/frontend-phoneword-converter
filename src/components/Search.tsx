import Button from './Button'
import {ChangeEvent, useEffect, useRef} from 'react';

type SearchProps = {
    input:string
    validInput: boolean
    onChange:(event:ChangeEvent) => void
    onConvert:(inputNumber:string) => void 
}

const Search = ({input, validInput, onChange, onConvert}:SearchProps) => {
    const inputRef = useRef<null | HTMLElement>(null)

    useEffect(() => {
        if (inputRef.current)
            inputRef.current.focus()
    }, [])

    return (
        <div className="Search">
            <h3>Input:</h3>
            <input type="text" value={input} onChange={onChange} ref={inputRef as any}/>
            <Button className='button' disabled={!validInput} onClick={() => onConvert(input)}>
                Convert
            </Button>
        </div>
    )
}

export default Search
