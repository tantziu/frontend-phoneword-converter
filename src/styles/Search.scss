.Search {
    margin-top: 20px;

    .searchInput {
        padding: 10px;
        border-radius: 5px;
        border: 1px solid #dddddd;
        background: transparent;
        margin-right: 10px;
        width: 50%
    }

}
