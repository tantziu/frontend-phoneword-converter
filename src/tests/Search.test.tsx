import Search from '../components/Search'
import renderer, { ReactTestRenderer } from 'react-test-renderer'

describe('Search', () => {
    const searchProps = {
        input: "567", 
        validInput: false, 
        onChange: jest.fn(),
        onConvert: jest.fn()
    }

    let component:ReactTestRenderer

    beforeEach(() => {
        component = renderer.create(<Search {...searchProps} />)
    })

    it('renders the input field with its value', () => {
        const value = component.root.findByType('input').props.value
        expect(value).toEqual('567')
    })

    it('changes the input field', () => {
        const pseudoEvent = 234

        component.root.findByType('input').props.onChange(pseudoEvent)

        expect(searchProps.onChange).toHaveBeenCalledTimes(1)
        expect(searchProps.onChange).toHaveBeenCalledWith(pseudoEvent)
    })

    it('disables the button and prevents submit', () => {
        component.update(
            <Search {...searchProps} input="" />
        )

        expect(component.root.findByType('button').props.disabled).toBeTruthy()
    })

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Search {...searchProps} />
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    }) 
})
