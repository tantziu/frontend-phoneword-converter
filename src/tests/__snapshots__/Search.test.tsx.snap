// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has a valid snapshot 1`] = `
<div
  className="Search"
>
  <input
    className="searchInput"
    onChange={[MockFunction]}
    placeholder="Type a number..."
    type="text"
    value="567"
  />
  <button
    className="Button disabled"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Convert
  </button>
</div>
`;
